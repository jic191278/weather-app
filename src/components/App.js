import React from "react";

import "../styles/app.css";
import forecastData from "../data/forecastData.json";

import LocationDetails from "./LocationDetails";

const App = () => {
  return (
    <div className="app">
      <h1>Weather App</h1>
      <LocationDetails location={forecastData.location} />
    </div>
  );
};

export default App;
