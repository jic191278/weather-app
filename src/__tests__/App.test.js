import React from 'react';
import { render, screen } from '@testing-library/react';
import App from '../components/App';

describe('App', () => {
  test('Render page correctly', () => {
    const { asFragment } = render(<App />);

    expect(asFragment()).toMatchSnapshot();
  });

  test('App header displays correctly', () => {
    render(<App />);

    const h1Element = screen.getByText(/weather app/i);
    expect(h1Element).toBeInTheDocument();
  });
});
