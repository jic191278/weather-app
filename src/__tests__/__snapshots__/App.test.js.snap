// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Render page correctly 1`] = `
<DocumentFragment>
  <div
    class="app"
  >
    <h1>
      Weather App
    </h1>
    <div
      class="location-details"
    >
      <h2>
        Manchester, UK
      </h2>
    </div>
  </div>
</DocumentFragment>
`;
